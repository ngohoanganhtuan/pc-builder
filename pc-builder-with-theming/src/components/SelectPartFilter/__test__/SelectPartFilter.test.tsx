import { render, fireEvent } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { SelectPartFilter } from '@components';

// Enums
import { BRAND_NAMES } from '@enums';

describe('Test [SelectPartFilter] component', () => {
  const handleMinChanged = jest.fn();
  const handleMaxChanged = jest.fn();
  const handleCheckboxChanged = jest.fn();
  const initialProps = {
    brands: [BRAND_NAMES.ADATA],
    selectedBrands: [BRAND_NAMES.ADATA],
    minPrice: 1000,
    maxPrice: 5000,
    handleMinChanged,
    handleMaxChanged,
    handleCheckboxChanged,
  };

  afterEach(() => {
    handleMinChanged.mockClear();
    handleMaxChanged.mockClear();
    handleCheckboxChanged.mockClear();
  });

  test('Component [SelectPartFilter] should render correctly', () => {
    const component = renderer.create(<SelectPartFilter {...initialProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [SelectPartFilter] calls function checkbox changed when user tick checkbox', () => {
    const { getAllByRole } = render(<SelectPartFilter {...initialProps} />);
    fireEvent.click(getAllByRole('checkbox')[0]);
    expect(handleCheckboxChanged).toHaveBeenCalled();
  });

  test('Component [SelectPartFilter] calls function min changed when user change min input', () => {
    const { getByTestId } = render(<SelectPartFilter {...initialProps} />);
    fireEvent.change(getByTestId('min-range-input'), {
      target: { value: '500' },
    });
    expect(handleMinChanged).toHaveBeenCalled();
  });

  test('Component [SelectPartFilter] calls function max changed when user change max input', () => {
    const { getByTestId } = render(<SelectPartFilter {...initialProps} />);
    fireEvent.change(getByTestId('max-range-input'), {
      target: { value: '1000' },
    });
    expect(handleMaxChanged).toHaveBeenCalled();
  });
});
