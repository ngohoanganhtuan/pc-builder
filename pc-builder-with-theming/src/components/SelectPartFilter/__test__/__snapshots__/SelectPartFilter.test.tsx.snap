// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [SelectPartFilter] component Component [SelectPartFilter] should render correctly 1`] = `
<div
  className="d-flex flex-column filter"
>
  <div
    className="mb-3"
  >
    <p
      className="form-label filter-brand"
    >
      Brand
    </p>
    <div
      className="d-flex flex-column flex-wrap"
    >
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-input"
          id="brand-checkbox-0"
          name="brand-checkbox-0"
          onChange={[MockFunction]}
          type="checkbox"
          value="Adata"
        />
        <label
          className="form-check-label brand-checkbox-label"
          htmlFor="brand-checkbox-0"
        >
          Adata
        </label>
      </div>
    </div>
  </div>
  <div
    className="mb-3"
  >
    <p
      className="form-label filter-price"
    >
      Price range
    </p>
    <div
      className="range-slider d-flex flex-row justify-content-between align-items-center gap-2"
    >
      <input
        className="form-control"
        data-testid="min-range-input"
        onChange={[MockFunction]}
        placeholder="Minimum"
        value={1000}
      />
      ~
      <input
        className="form-control"
        data-testid="max-range-input"
        onChange={[MockFunction]}
        placeholder="Maximum"
        value={5000}
      />
    </div>
  </div>
</div>
`;
